editor/src/core/workers/parser-printer/uid-fix.ts

```diff
@@ -133,6 +133,17 @@ function updateUID<T>(
   fixUIDsState: FixUIDsState,
   baseValue: T,
 ): T {
+  function addMapping(originalUID: string, newUID: string): void {
+    const priorMapping = fixUIDsState.mappings.find((mapping) => {
```

Comment:
Won't `find` here cause perf problems? I'm not sure this will scale with large projects

Reply:
There doesn't tend to be huge numbers of mappings and this runs on the parser-printer worker so the performance demands are much lower than say during a canvas operation.

Reply:
Also these mappings are per file, so the size of the project doesn't influence this particularly.

Reply:
I think this is definitely worth checking for a large project. It seems we now use the `mappings` in two places: one which calls `find` based on the `mapping.originalUID`, and now this one which calls `fine` based on `mapping.newUID`. If this is a performance issue maybe it would be better two create a pair of maps (from originalUID to newUID and vice versa)?

Reply:
Urgh, I've just seen your comments now that I've posted that. When I posted it they weren't showing because AJAX is still pretty new technology

Reply:
Oh yes and I tested this on the Before I Go and the 31x10 performance test projects and performance of the updates coming in appeared to be fine.